internal/handlers/images.go

```diff
@@ -94,15 +112,73 @@ func (h *ImageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	defer f.Close()
 
-	fileInfo, err := f.Stat()
+	isoReader, err := h.imageStreamForID(clusterID, f)
 	if err != nil {
-		log.Errorf("Error getting file info: %v\n", err)
+		log.Errorf("Error creating image stream: %v\n", err)
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
+	//TODO: set modified time correctly (MGMT-7274)
+
 	fileName := fmt.Sprintf("%s-discovery.iso", clusterID)
-	http.ServeContent(w, r, fileName, fileInfo.ModTime(), f)
+	http.ServeContent(w, r, fileName, time.Now(), isoReader)
+}
+
+func (h *ImageHandler) imageStreamForID(imageID string, baseISOPath string) (io.ReadSeeker, error) {
+	ignition, err := h.ignitionContent(imageID)
+	if err != nil {
+		return nil, err
+	}
+	return h.GenerateImageStream(baseISOPath, ignition)
+}
+
+func (h *ImageHandler) ignitionContent(imageID string) ([]byte, error) {
```

Comment:
Is this something that should be done outside the handler? I had expected `imageStreamForID` to call something internal to grab it from the assisted-service.

Reply:
I'm not sure, I was thinking about creating some kind of "client" for this, but didn't feel like it was worth it just yet. Maybe I'll refactor when I add the minimal iso and initrd query.